Add redis metrics cralwer plugins

tatsuhirochiba opened the pull request:
This PR supports to gather redis provided metrics on localhost and container. 
The detail is described at issue #196. 

Signed-off-by: Tatsuhiro Chiba chiba@jp.ibm.com

sahilsuneja1 commented:
@tatsuhirochiba, I think travis failed to run the tests since your commits weren't 'signed'- required by @ricarkol

ricarkol commented:
@tatsuhirochiba , you need to commit your changes with `-s`, like this: `commit -s -m "change summary"`

ricarkol commented on requirements-app.txt:
```diff
@@ -0,0 +1 @@
+redis>=2.10.5
```
Would prefer to move this to requirements.txt, as it is not only necessary for tests.

ricarkol commented on crawler/plugins/applications/util.py:
```diff
@@ -0,0 +1,67 @@
+    '''
+
+    def __init__(self, container_id):
+        self.client = docker.Client(
```
You can use dockerutils.py for these kind of calls.

tatsuhirochiba commented:
@sahilsuneja1 @ricarkol Thank you for checking! I fixed to add singed message to all of my commits. Also I changed to use `dockerutil module` for retrieving docker inspect information.

ricarkol commented on crawler/plugins/applications/util.py:
```diff
@@ -0,0 +1,63 @@
+
+    def __init__(self, container_id):
+        self.inspect = dockerutils.exec_dockerinspect(container_id)
+        # self.cotainers = dockerutils.exec_dockerlistcontainers()
```
You might have left this comment by mistake.

Would you prefer to add these methods: `is_app_container`, `get_ip`, `get_ports` to the DockerContainer class (dockercontainer.py)?

tatsuhirochiba replied:
OK, I will remove it.

I prefer to add `get_ip` and `get_ports` methods in DockerContainer class. `is_app_container` method is not needed to add actually. Instead, I plan to use `dockercontainer.image` variable.  

So can I include these two methods? And then, I prefer to use `dockercontainer class` or `dockerutils.py` instead of `app/util.py`.

sahilsuneja1 commented:
'--features application'  that's no longer needed right?

tatsuhirochiba commented:
@sahilsuneja1 Yes, it is ok for me to use `--plugins` only, instead of using `--feature`.  
In current implementation, however, It seems that we needs to specify both of `feature` and `plugins` to enable user specified plugin, so I am using `feature` argument now.

sahilsuneja1 commented:
@tatsuhirochiba: yes, going forward we would like plugins to be present inside `crawler.conf`'s `enabled_plugins`. However, we will still support `--features` for legacy reasons, so if an argument with `--features` matches a plugin's `get_feature()` output, the plugin will be picked up by the crawler.
